config/initializers/0_clowder-config.rb

```diff
@@ -19,13 +19,27 @@
   redis_url = "#{config.dig('inMemoryDb', 'hostname')}:#{config.dig('inMemoryDb', 'port')}"
   redis_password = config.dig('inMemoryDb', 'password')
 
+  # Kafka
+  first_kafka_server_config = config.dig('kafka', 'brokers')[0]
+
+  kafka_server_config = {
+    brokers: "#{first_kafka_server_config&.dig('hostname')}:#{first_kafka_server_config&.dig('port')}",
+    security_protocol: first_kafka_server_config.dig('authtype')
+  }
+
+  if kafka_server_config.security_protocol == 'sasl_ssl'
```

Comment:
Use safe navigation

```suggestion
  if kafka_server_config&.security_protocol == 'sasl_ssl'
```

Reply:
ugh - I didn't see this suggestion and went with `kafka_server_config['security_protocol']` syntax :/ 

What is recommended in Rails better ?

Reply:
Both should work and return `nil` if the key doesn't exist. The suggested syntax is a newer Ruby feature for safe navigation. In hashes, there is also `dig` which allows multiple levels, like `my_hash.dig('foo', 'bar', 'baz')` and will return `nil` if any key along the way doesn't exist.